File: 035_Search_Insert_Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        nums.append(float('Inf'))
        length = len(nums)
        low = 0
        top = length-1
        
        while top is not low:
            n = (top-low)/2
            if target < nums[low+n]:
                top = low+n
            elif target > nums[low+n+1]:
                low = low+n+1
            elif target is nums[low+n]:
                return low+n
            else:
                return low+n+1
        
        return low

File: 035_Search_Insert_Position.cpp
#include<limits.h>
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        vector<int> list = nums;
        list.push_back(INT_MAX);
        
        int n;
        int low = 0;
        int top = list.size()-1;
        
        while(low != top){
            
            n = (top-low)/2;
            if(list[low+n] > target)
                top = low+n;
            else if(list[low+n+1] < target)
                low = low+n+1;
            else if(list[low+n] == target)
                return low+n;
            else
                return low+n+1;
            
        }
        
        return low;
        
    }
};
